service/src/test/kotlin/app/cash/backfila/actions/StartStopBackfillActionTest.kt

```diff
@@ -27,128 +30,162 @@ import misk.hibernate.load
 import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Nested
 import org.junit.jupiter.api.Test
+import javax.inject.Inject
+import kotlin.test.assertNotNull
 
 @MiskTest(startService = true)
 class StartStopBackfillActionTest {
   @Suppress("unused")
-  @MiskTestModule
-  val module: Module = BackfilaTestingModule()
+  @MiskTestModule val module: Module = BackfilaTestingModule()
 
-  @Inject
-  lateinit var configureServiceAction: ConfigureServiceAction
+  @Inject lateinit var configureServiceAction: ConfigureServiceAction
 
-  @Inject
-  lateinit var createBackfillAction: CreateBackfillAction
+  @Inject lateinit var createBackfillAction: CreateBackfillAction
 
-  @Inject
-  lateinit var startBackfillAction: StartBackfillAction
+  @Inject lateinit var startBackfillAction: StartBackfillAction
 
-  @Inject
-  lateinit var stopBackfillAction: StopBackfillAction
+  @Inject lateinit var stopBackfillAction: StopBackfillAction
 
-  @Inject
-  lateinit var getBackfillRunsAction: GetBackfillRunsAction
+  @Inject lateinit var getBackfillRunsAction: GetBackfillRunsAction
 
-  @Inject
-  lateinit var getBackfillStatusAction: GetBackfillStatusAction
+  @Inject lateinit var getBackfillStatusAction: GetBackfillStatusAction
 
-  @Inject
-  lateinit var queryFactory: Query.Factory
+  @Inject lateinit var queryFactory: Query.Factory
 
-  @Inject
-  lateinit var scope: ActionScope
+  @Inject lateinit var scope: ActionScope
 
-  @Inject
-  @BackfilaDb
-  lateinit var transacter: Transacter
+  @Inject @BackfilaDb lateinit var transacter: Transacter
 
-  @Test
-  fun startAndStop() {
+  @Inject lateinit var fakeBackfilaClientServiceClient: FakeBackfilaClientServiceClient
+
+  @BeforeEach
+  fun setup() {
+    // Always configure the backfills before any test
     scope.fakeCaller(service = "deep-fryer") {
       configureServiceAction.configureService(
         ConfigureServiceRequest.Builder()
           .backfills(
             listOf(
-              ConfigureServiceRequest.BackfillData(
-                "ChickenSandwich", "Description", listOf(), null,
-                null, false
-              )
+              ConfigureServiceRequest.BackfillData.Builder()
+                .name("ChickenSandwich")
+                .description("Description")
+                .build(),
+              ConfigureServiceRequest.BackfillData.Builder()
+                .name("BeefSandwich")
+                .description("Description")
+                .build()
             )
           )
           .connector_type(Connectors.ENVOY)
           .build()
       )
     }
+  }
+
+  @Test fun `Backfila starts with no backfills`() {
     scope.fakeCaller(user = "molly") {
       var backfillRuns = getBackfillRunsAction.backfillRuns("deep-fryer")
-      assertThat(backfillRuns.paused_backfills).hasSize(0)
-      assertThat(backfillRuns.running_backfills).hasSize(0)
+      softAssert {
```

Comment:
what? why?

Reply:
soft assert gives you all the errors instead of just the first one.

Reply:
hmm ok